packages/ui/src/components/ManageKafkaPermissions/components/PermissionsTable.tsx

```diff
@@ -58,12 +58,12 @@ export const PermissionsTable = <T extends Permissions>({
     permission: t("table.permissions_column_title"),
     resource: t("table.resource_column_title"),
   };
-  const onCheck = (isSelecting: boolean, rowIndex: number) => {
-    if (rowIndex != undefined) {
+  const onCheck = (row: { rowIndex: number }, isSelecting: boolean) => {
```

Comment:
```suggestion
  const onCheck: TableViewProps['onCheck'] = (row, isSelecting) => {
```

and also, shouldn't that be wrapped in a useCallback?